blues/airnote.live
# feat: move Airnote quickstart docs into Airnote site

# Problem Context

Take the MDX files for the quickstart in the dev.blues site and move them over to the airnote.live site.

These files can live under the “Setup Guide” section. For a first pass, just get the documentation to the Airnote site, then we can work on formatting and styling.

Also update the redirects that go to dev.blues, like start.airnote.live. 

## Changes

* Add new Quickstart link to site header links (this is always visible regardless of whether a user has an Airnote device URL in their browser or not)
* Add all the MDX files and images for Airnote setup, air quality info, faqs inside the new Quickstart section
* Add IDs and anchor links to all headers inside MDX files so users can click on titles or docs can deep link to specific sections in the markdown and jump the user to that point in the quickstart docs
* Add new `Note` component to display stylized callout notes inside of MDX docs (very similar to styled Notes in dev.blues site)
* Add new error scenario for site in case MDX files can't load 
* Adjust all links in site that were directing to `start.airnote.live` to now direct internally to `/quickstart` which is where the getting started docs now reside and update language in links to reflect that
* Update any old links going to discuss.blues.io or blues.io to now go to .com versions of each site
* New CSS to style markdown images, tables, and blockquotes
* New e2e to test the MDX files are rendering correctly
* Update redirects file to ensure start.airnote.live now redirects to quickstart docs within Airnote site

## Screenshot (if applicable)

![Screenshot 2023-12-13 at 10 32 16 AM](https://github.com/blues/airnote.live/assets/20400845/ec110462-d654-4863-a9a9-8bbca7241f6b)

## Testing

Unit / integration / e2e tests?

All unit tests pass.
New e2e added to check all sections of the quickstart MDX files render correctly.

Steps to test manually?

1. Go to http://localhost:5173
2. See "Quickstart" link in header, click it
3. Inside of Quickstart section see all the different pieces of setting up a new Airnote
4. Make sure all Notes look good, block quotes, tables, images. Click on header links and see the page scroll smoothly to anchor links 
5. Resize the page and see the images scale up and down, make sure nothing overruns the page width.

Any other sorts of testing notes to include?

## Any other related PRs

dev.blues - https://github.com/blues/blues.dev/pull/1495

## Ticket(s)

https://trello.com/c/rHYj3Dw6/546-move-quickstart-docs-from-devblues-to-airnotelive-site
